src/components/NavBar.jsx

```diff
@@ -1,17 +1,53 @@
-import * as React from "react";
+import React, { useState } from "react";
 import { Tab, Box } from "@mui/material";
 import { useNavigate } from "react-router-dom";
-import { Button } from "@mui/base";
+import { logoutAccount } from "@/common/ServerAPI"
+import { useGlobalStore } from "@/common/useGlobalStore";
+import SnackBar from "./SnackBar";
 
 export default function NavBar() {
   const navigate = useNavigate();
+  const symmetricKey = useGlobalStore((state) => state.symmetricKey);
+  const { deleteSymmetricKey } = useGlobalStore();
+  const [snackBarMessaage, setSnackBarMessage] = useState("")
+  const [openSnackbar, setOpenSnackbar] = useState(false);
+
+  const handleLogout = async () => {
+    try {
+      await logoutAccount();
+      deleteSymmetricKey();
+      localStorage.removeItem("key-fortress");
+      navigate("/login-signup");
+    } catch(err) {
+      setSnackBarMessage("Please sign in and try again.")
+      setOpenSnackbar(true);
+      navigate("/login-signup");
```

Comment:
I notice that (similar to what I experienced in VaultItemTiles) if the API call returns an error, the user is redirected before they can see the Snackbar. If you want it to be visible you'd have to [use a setTimeout to cause a delay](https://github.com/secure-password-manager/PasswordManagerWeb/blob/23f5538656b811642b9851f59b8e46f2c2b2820f/src/components/VaultItemTiles.jsx#L42).

Reply:
I tried the setTimeout and I am not sure why it doesn't work in this case. Say a user goes into the application settings to delete their session cookie or csrf token, and then they get a 403 response. Is there anything wrong with not notifying the of the error and navigate them to login-signup? Since they don't have a session cookie anymore anyways they wouldn't be able to access their vault.